test/src/module/common/timeTest.c

```diff
@@ -31,5 +32,51 @@ testRun(void)
     }
 
+    // *****************************************************************************************************************************
+    if (testBegin("yearIsLeap(), dayOfYear(), timePartsValid(), and epochFromParts()"))
+    {
+        TEST_TITLE("is leap year");
+
+        TEST_RESULT_BOOL(yearIsLeap(1996), true, "1996 is leap year");
+        TEST_RESULT_BOOL(yearIsLeap(2000), true, "2000 is leap year");
+        TEST_RESULT_BOOL(yearIsLeap(2001), false, "2001 is not leap year");
+        TEST_RESULT_BOOL(yearIsLeap(2100), false, "2100 is not leap year");
+
+        // -------------------------------------------------------------------------------------------------------------------------
+        TEST_TITLE("day of year");
+
+        TEST_RESULT_INT(dayOfYear(1995, 3, 1), 60, "1995-03-01 day of year");
+        TEST_RESULT_INT(dayOfYear(1996, 3, 1), 61, "1996-03-01 day of year");
+
+        // -------------------------------------------------------------------------------------------------------------------------
+        TEST_TITLE("date parts valid");
+
+        TEST_ERROR(datePartsValid(1969,  0,  0), FormatError, "invalid date 1969-00-00");
+        TEST_ERROR(datePartsValid(1970,  0,  0), FormatError, "invalid date 1970-00-00");
+        TEST_ERROR(datePartsValid(1970, 13,  0), FormatError, "invalid date 1970-13-00");
+        TEST_ERROR(datePartsValid(1970, 12,  0), FormatError, "invalid date 1970-12-00");
+        TEST_ERROR(datePartsValid(1970,  2, 29), FormatError, "invalid date 1970-02-29");
+        TEST_RESULT_VOID(datePartsValid(1972, 2, 29), "valid date");
+
+        // -------------------------------------------------------------------------------------------------------------------------
+        TEST_TITLE("time parts valid");
+
+        TEST_ERROR(timePartsValid(-1,  0,  0), FormatError, "invalid time -1:00:00");
+        TEST_ERROR(timePartsValid(24,  0,  0), FormatError, "invalid time 24:00:00");
+        TEST_ERROR(timePartsValid( 0, -1,  0), FormatError, "invalid time 00:-1:00");
+        TEST_ERROR(timePartsValid( 0, 60,  0), FormatError, "invalid time 00:60:00");
+        TEST_ERROR(timePartsValid( 0,  0, -1), FormatError, "invalid time 00:00:-1");
+        TEST_ERROR(timePartsValid( 0,  0, 60), FormatError, "invalid time 00:00:60");
+        TEST_RESULT_VOID(timePartsValid(23, 59, 59), "valid time");
+
+        // -------------------------------------------------------------------------------------------------------------------------
+        TEST_TITLE("time from parts");
+
+        TEST_RESULT_INT(epochFromParts(1970, 1, 1, 0, 0, 0), 0, "beginning of epoch");
+        TEST_RESULT_INT(epochFromParts(1996, 3, 3, 3, 3, 3), 825822183, "march of leap year");
+        TEST_RESULT_INT(epochFromParts(2000, 4, 4, 4, 4, 4), 954821044, "april of non-leap year");
+        TEST_RESULT_INT(epochFromParts(2038, 1, 19, 3, 14, 7), INT_MAX, "end of 32-bit signed int epoch");
```

Comment:
So do we need to somehow handle or throw an error for 32-bit systems after the Jan 19, 2038 date?

Reply:
We'll worry about that in 18 years.  Probably we'll just disallow builds on systems with 32-bit time_t.

Note that this is not a backrest issue  -- everything will work with a properly sized time_t.